### PR TITLE
Filter template ids in popup menu entries

### LINKED ISSUE #3627: Filter template ids in modeling tracking
### What should we do?

The id of popup menu entries is tracked and template entry ids are added as `replace/append.<template-id>` (in `camunda-bpmn-js`). We should interpret that id and send as `templateId: <template-id>`.

### Why should we do it?

Part of https://github.com/bpmn-io/bpmn-js-tracking/issues/19

### PR DESCRIPTION
**Pull request by smbea**
Also closes a whole bunch of additional _fixed upstream_ issue through dependency bumps.

Closes https://github.com/camunda/camunda-modeler/issues/3627
Closes https://github.com/camunda/camunda-modeler/issues/3510
Closes https://github.com/camunda/camunda-modeler/issues/3599
Closes https://github.com/camunda/camunda-modeler/issues/3435
Closes https://github.com/camunda/camunda-modeler/issues/3568
Closes https://github.com/camunda/camunda-modeler/issues/3602
Closes https://github.com/camunda/camunda-modeler/issues/3569
Closes https://github.com/camunda/camunda-modeler/issues/3555

### CODE REVIEW

File: client/src/plugins/user-journey-statistics/event-handlers/__tests__/ModelingEventHandlerSpec.js [hunk 1]
```diff
@@ -85,25 +85,84 @@ describe('<ModelingEventHandler>', () => {
...
-              elementArray: [ create('shape') ]
+        expect(track).to.have.been.calledWith('some:event', {
+          nonAnElement: 'someValue',
+          element: {},
+          elementArray: [ {} ]
+        });
+      });
+
+
+      describe('popupmenu.Trigger - element templates id', () => {
```
Comment (nikku) [inline client/src/plugins/user-journey-statistics/event-handlers/__tests__/ModelingEventHandlerSpec.js:113]:
```suggestion
      describe('popupMenu.trigger - element templates id', () => {
```

Is that what we want to express?

Comment (smbea) [reply client/src/plugins/user-journey-statistics/event-handlers/__tests__/ModelingEventHandlerSpec.js]: Yes, typo on my end

File: client/package.json [hunk 2]
```diff
@@ -28,8 +28,8 @@
     "bpmn-js": "^13.0.0",
     "bpmn-js-properties-panel": "^1.22.1",
     "bpmn-moddle": "^8.0.1",
-    "bpmn-js-tracking": "0.2.0",
-    "camunda-bpmn-js": "^2.3.1",
+    "bpmn-js-tracking": "0.3.0",
+    "camunda-bpmn-js": "^2.5.0",
```
Comment (nikku) [inline client/package.json:32]: Let's make this bump explicit, and include changes pulled in via `camunda-bpmn-js` bump in the `CHANGELOG`.

**Comment by nikku** [comment]
@smbea I made this a full dependency bump, including properties panel and linking.

Please ensure the bump commits actually reference the issues they close, cf. https://github.com/camunda/camunda-modeler/pull/3633/commits/4faca2c8d1980f81ef9b6ab839c6740d2e683dc9 for example.

I'll also follow-up with the team regarding this; let's ensure it is clear (traceable) what upstream changes actually _upstream fix_ a certain issue.

**Review by nikku — APPROVED** [review]

**Pull request merged**